File: 1306-minimum-absolute-difference/minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        
        List<List<Integer>> ans = new ArrayList<>();

        Arrays.sort(arr);

        int min_diff = Integer.MAX_VALUE;
        for(int i=1;i<arr.length;i++)
        {
            int curr = Math.abs(arr[i]-arr[i-1]);

            min_diff = Math.min(min_diff,curr);
        }
        
        for(int i=1;i<arr.length;i++)
        {
            int curr = Math.abs(arr[i]-arr[i-1]);

            if(curr == min_diff)
            {
                List<Integer> curr1 = new ArrayList<>();
                curr1.add(arr[i-1]);
                curr1.add(arr[i]);

                ans.add(curr1);
            }
        }

        return ans;



    }
}

File: 1306-minimum-absolute-difference/minimum-absolute-difference.cpp
class Solution {
public:
    
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        
        sort(arr.begin(),arr.end());

        int min_abs_diff = INT_MAX;
        for(int i=1;i<arr.size();i++)
        {
            min_abs_diff = min(min_abs_diff,arr[i]-arr[i-1]);
        }
        
        vector<vector<int>>answer;
        for(int i=1;i<arr.size();i++)
        {
            int curr_diff = arr[i]-arr[i-1];
            if(curr_diff == min_abs_diff)
                answer.push_back({arr[i-1],arr[i]});

        }

        return answer;
    }
};